### PR TITLE
Pr9115 android signing uplift 1.7.x

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
## Submitter Checklist:
Uplift for https://github.com/brave/brave-browser/pull/9115
- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] If adding a new directory with NPM packages, ensure this directory is
  added to `scripts/audit.js`.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by kjozwiak** [comment]
Restarting `Android` as `build` failed as per https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/1/execution/node/279/log/:

```
14:11:49  ERROR at //build/timestamp.gni:31:19: Script returned non-zero exit code.
14:11:49  build_timestamp = exec_script(compute_build_timestamp,
14:11:49                    ^----------
14:11:49  Current dir: /home/ubuntu/jenkins/workspace/brave-browser-build-pr_PR-9116/src/out/android_Release_arm/
14:11:49  Command: python /home/ubuntu/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py official
14:11:49  Returned 1.
14:11:49  stderr:
14:11:49  
14:11:49  Traceback (most recent call last):
14:11:49    File "/home/ubuntu/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py", line 127, in <module>
14:11:49      sys.exit(main())
14:11:49    File "/home/ubuntu/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py", line 113, in main
14:11:49      last_commit_timestamp = int(open(lastchange_file).read())
14:11:49  IOError: [Errno 2] No such file or directory: '/home/ubuntu/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/util/LASTCHANGE.committime'
14:11:49  
14:11:49  See //base/BUILD.gn:34:1: whence it was imported.
14:11:49  import("//build/timestamp.gni")
14:11:49  ^-----------------------------
14:11:49  See //BUILD.gn:318:7: which caused the file to be included.
14:11:49        "//base:base_junit_tests",
14:11:49        ^------------------------
```

Restarting `iOS` as `init` failed as per https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/1/execution/node/156/log/.

Restarting `Linux` as `build` failed as per https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/1/execution/node/335/log/:

```
14:12:41  ERROR at //build/config/sysroot.gni:57:5: Assertion failed.
14:12:41      assert(
14:12:41      ^-----
14:12:41  Missing sysroot (//build/linux/debian_sid_amd64-sysroot). To fix, run: build/linux/sysroot_scripts/install-sysroot.py --arch=amd64
14:12:41  See //build/config/sysroot.gni:58:9: 
14:12:41          exec_script("//build/dir_exists.py",
14:12:41          ^-----------------------------------
14:12:41  This is where it was set.
14:12:41  See //chrome/installer/BUILD.gn:7:1: whence it was imported.
14:12:41  import("//build/config/sysroot.gni")
14:12:41  ^----------------------------------
14:12:41  See //BUILD.gn:71:5: which caused the file to be included.
14:12:41      "//chrome/installer",
```

Restarting `macOS` as `build` failed as per https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/1/execution/node/507/log/:

```
14:27:25  ERROR at //build/timestamp.gni:31:19: Script returned non-zero exit code.
14:27:25  build_timestamp = exec_script(compute_build_timestamp,
14:27:25                    ^----------
14:27:25  Current dir: /Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-9116/src/out/Release/
14:27:25  Command: python /Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py official
14:27:25  Returned 1.
14:27:25  stderr:
14:27:25  
14:27:25  Traceback (most recent call last):
14:27:25    File "/Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py", line 127, in <module>
14:27:25      sys.exit(main())
14:27:25    File "/Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/compute_build_timestamp.py", line 113, in main
14:27:25      last_commit_timestamp = int(open(lastchange_file).read())
14:27:25  IOError: [Errno 2] No such file or directory: '/Users/jenkins/jenkins/workspace/brave-browser-build-pr_PR-9116/src/build/util/LASTCHANGE.committime'
14:27:25  
14:27:25  See //base/BUILD.gn:34:1: whence it was imported.
14:27:25  import("//build/timestamp.gni")
14:27:25  ^-----------------------------
14:27:25  See //BUILD.gn:68:5: which caused the file to be included.
14:27:25      "//base:base_perftests",
14:27:25      ^----------------------
```

Restarting `Win` as `build` failed as per https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/1/execution/node/395/log/:

```
14:19:59  Traceback (most recent call last):
14:19:59  
14:19:59    File "C:/.7.x/src/build/compute_build_timestamp.py", line 127, in <module>
14:19:59  
14:19:59      
14:19:59  sys.exit(main())
14:19:59  
14:19:59    File "C:/.7.x/src/build/compute_build_timestamp.py", line 113, in main
14:19:59  
14:19:59  
14:19:59      
14:19:59  last_commit_timestamp = int(open(lastchange_file).read())
14:19:59  
14:19:59  IOErro
14:19:59  r: [Errno 2] No such file or directory: 'C:\\.7.x\\src\\build\\util\\LASTCHANGE.committime'
14:19:59  
14:19:59  ERROR at //build/timestamp.gni:31:19: Script returned non-zero exit code.
14:19:59  build_timestamp = exec_script(compute_build_timestamp,
14:19:59                    ^----------
14:19:59  Current dir: C:/.7.x/src/out/Release/
14:19:59  Command: C:/.7.x/vendor/depot_tools/bootstrap-3_8_0_chromium_8_bin/python/bin/python.exe C:/.7.x/src/build/compute_build_timestamp.py official
14:19:59  Returned 1.
14:19:59  See //base/BUILD.gn:34:1: whence it was imported.
14:19:59  import("//build/timestamp.gni")
14:19:59  ^-----------------------------
14:19:59  See //BUILD.gn:69:5: which caused the file to be included.
14:19:59      "//base:base_unittests",
14:19:59      ^----------------------
```

**Comment by bridiver** [comment]
cc @mihaiplesa - I believe this should be fixed by https://github.com/brave/brave-browser/pull/9112/files

**Comment by bridiver** [comment]
actually it looks like it's still  a problem without depth=2 cc @linhkikuchi

**Comment by bsclifton** [comment]
Moving to 1.7.x Hotfix 1 milestone; let's not merge until after the 1.7 release 😄

**Comment by SergeyZhukovsky** [comment]
Moving back to 1.7.x as per DM discussion. Android builds are blocked without that.

**Comment by bsclifton** [comment]
I'm seeing an error here during the build step
https://ci.brave.com/job/brave-browser-build-pr/job/PR-9116/3/execution/node/234/log/

Since this is critical for Android, I'm going to replay the CI for Android only to make sure it works. Rebased after merging https://github.com/brave/brave-browser/pull/9156 - it should restart in a minute here

File: lib/createDist.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ const createDist = (buildConfig = config.defaultBuildConfig, options) => {
   fs.removeSync(path.join(config.outputDir, 'dist'))
   config.buildTarget = 'create_dist'
   util.buildTarget()
+  if (config.targetOS === 'android') {
```
Comment (bridiver) [inline lib/createDist.js:22]: this should still have `config.shouldSign()` in addition to the targetOS check

Comment (SergeyZhukovsky) [reply lib/createDist.js:22]: we can add it as a follow up, as should be applied in any case on all 3 branches. But I think we shouldn't skip signing Android apks. First it's fast, second we will need that on tests to check upgrade scenarios.

**Pull request merged**